src/Akeneo/Pim/Enrichment/Bundle/Command/BackoffElasticSearchStateHandler.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+
+namespace Akeneo\Pim\Enrichment\Bundle\Command;
+
+use Elasticsearch\Common\Exceptions\BadRequest400Exception;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+class BackoffElasticSearchStateHandler
+{
+    public const RETRY_COUNTER = 10;
+    public const INITIAL_WAIT_DELAY = 10;
+    public const BACKOFF_LOGARITHMIC_INCREMENT = 2;
+
+    public function doIndex(iterable $chunkedCodes, ProgressBar $progressBar, \Closure $codesEsHandler): int
+    {
+        $indexedCount = 0;
+
+        $progressBar->start();
+        foreach ($chunkedCodes as $productModelCodes) {
```

Comment:
```suggestion
        foreach ($chunkedCodes as $codes) {
```

Could be productModel agnostic